README
NAME
    Perl6::Doc - Perl 6 documentation

SYNOPSIS
    This distribution is documentation-only. It contains no code for you to
    run. Included are various Perl 5 Pod and Perl 6 Pod files describing the
    Perl 6 language. For a tool to read this documentation, see grok.

DESCRIPTION
    Currently included in this distribution:

  Design docs
   Apocalypses (outdated)
    Larry Wall started the Apocalypse (latin for revelation) series as a
    systematic way of answering the RFCs (Request For Comments) that started
    the design process for Perl 6. Each Apocalypse corresponds to a chapter
    in the book *Programming Perl, 3rd edition*, and addresses the features
    relating to that chapter in the book that are likely to change.

    Larry addresses each relevant RFC, and gives reasons why he accepted or
    rejected various pieces of it. But each Apocalypse also goes beyond a
    simple "yes" and "no" response to attack the roots of the problems
    identified in the RFCs.

    Note: These documents are outdated and and will not be updated to match
    the specification (see "Synopses").

   Exegeses (outdated)
    Damian Conway's Exegeses (latin for *explanation*) are extensions of
    each Apocalypse. Each Exegesis is built around a practical code example
    that applies and explains the new ideas.

    Note: These documents are outdated and and will not be updated to match
    the specification (see "Synopses").

   Synopses
    The Synopsis (latin for comparison) started as a fast to read diff
    between Perl 5 and 6. Because they are also easier to maintain, all
    changes of the language, that are evolving from the design process are
    written down here first. The Apocalypses and Exegeses are frozen as
    "historic documents".

    In other words, these docs may change slightly or radically. But the
    expectation is that they are "very close" to the final shape of Perl 6.

    The Synopsis documents are to be taken as the formal specification for
    Perl 6 implementations, while still being reference documentation for
    Perl 6, like *Programming Perl* is for Perl 5.

    Note that while these documents still being subjected to the rigours of
    cross-examination through implementation.

  Magazine articles
    These are Pod-ified versions of magazine articles about Perl 6 that have
    appeared in print during the design process. Take these with a grain of
    salt as they might not describe Perl 6 as it is now. They are mostly
    from <http://www.perl.com> and <http://www.perl-magazin.de/>.

  Perl 6 Table Index
    A Pod-ified version of
    <http://www.perlfoundation.org/perl6/index.cgi?perl_table_index> is
    included.

  Man pages
    Currently included are draft versions of perlintro and perlsyn.

HISTORY
    As of version 0.40, this distribution has been overhauled and is no
    longer maintained by Herbert Breunung.

PACKAGING
    Hinrik Örn Sigurðsson, hinrik.sig@gmail.com

    I'm not the author of the documentation in this distro, of course, I
    merely maintain this package.

SOURCES
    All Apocalypses and Exegeses were taken from the official Perl
    development site: <http://dev.perl.org/perl6/>

    All Synopese were taken from the Pugs repository:
    <http://svn.pugscode.org/pugs/docs/Perl6/>

    The magazine articles are from perl.com:
    <http://perl.com/pub/q/Article_Archive#Perl%206>

LICENSE AND COPYRIGHT
    This copyright applies only to the "Perl6::Doc" Perl software
    distribution, not the documents bundled within.

    Copyright (c) 2009, Hinrik Örn Sigurðsson hinrik.sig@gmail.com.

    "grok" is distributed under the terms of the Artistic License 2.0. For
    more details, see the full text of the license in the file LICENSE that
    came with this distribution.

  Scribes
    These are the authors of the included docs, named in the order their
    work was added. This list is not exhaustive.

    * Larry Wall larry@wall.org

    * Damian Conway damian@conway.org

    * Luke Palmer luke@luqui.org

    * Allison Randal al@shadowed.net

    * Audrey Tang autrijus@cpan.org

    * Ingy döt Net ingy@cpan.org

    * Sam Vilain samv@cpan.org

    * Kirrily "Skud" Robert skud@cpan.org

    * Moritz Lenz moritz@fau2ik3.org

    * David Koenig karhu@u.washington.edu

    * Jonathan Scott Duff duff@pobox.com

    * Phil Crow philcrow2000@yahoo.com

    * chromatic chromatic@oreilly.com

    * Mark-Jason Dominus mjd@songline.com

    * Shmarya shmarya.rubenstein@gmail.com

    * Pawel Murias 13pawel@gazeta.pl

    * Herbert Breunung lichtkind@cpan.org



Changes
{{$NEXT}}

0.47  Mon Dec 6 03:25:57 GMT 2010
    - Convert S26-documentation.pod to UTF-8

0.46  Fri Dec 3 07:15:01 GMT 2010
    - Use Dist::Zilla
    - Fix Perl 6 tablet hyperlink link and some other stuff pointed out
      by Herbert Breunung
    - Update the Perl 6 tablets to the latest version
    - Update all Synopses (and add S08-capture)

0.45 Mon Aug 17 17:12:06 GMT 2009
    - Added perlobjintro and perlreintro

0.44 Wed Aug 12 16:44:05 GMT 2009
    - Various small fixes to man pages

0.43 Fri Jul 24 17:06:22 GMT 2009
    - Add a test file

0.42 Fri Jul 24 14:42:36 GMT 2009
    - Added first drafts of perlintro and perlsyn

0.41 Thu Jul 16 06:10:00 GMT 2009
    - Add Pod-ified version of the Perl 6 Table Index

0.40 Tue Jul 14 13:12:23 EDT 2009
    - First release of a revamped doc-only version of this distribution, now
      maintained by Hinrik Örn Sigurðsson


devel/parse_perl6_table_index.pl
use strict;
use warnings;

#autoflush STDOUT
$| = 1;

#
# Convert html back to text by removing html tags and converting
# html entities.
#
sub _to_text {
	my $text = shift;
	$text =~ s/<em>(.+?)<\/em>/$1/g;
	$text =~ s/<.+?>//g;
	$text =~ s/&amp;/&/g;
	$text =~ s/&lt;/</g;
	$text =~ s/&gt;/>/g;
	$text =~ s/&quot;/"/g;
	$text =~ s/&nbsp;/ /g;
	return $text;
}

#
# Converts Perl 6 table index help to POD after getting it from the web.
#
sub _perl6_table_index_to_pod {
	my $url = shift;
	
	print "Loading $url\n";
	require LWP::UserAgent;
	require HTTP::Request;
	my $ua = LWP::UserAgent->new;
	my $req = HTTP::Request->new(GET => $url);
	my $res = $ua->request($req);
	if(not $res->is_success) {
		die $res->status_line, "\n";
	}
	
	my $pod = <<"POD";
=head1 Perl 6 table index

This is the POD version of $url

=head1 AUTHORS

This POD was generated by Ahmad M. Zawawi <ahmad.zawawi\@gmail.com> via the tool:
http://svn.perlide.org/padre/trunk/Padre-Plugin-Perl6/parse_perl6_table_index.pl

For authors of the original wiki place, see:
http://www.perlfoundation.org/perl6/index.cgi?action=revision_list;page_name=perl_table_index

=head1 LICENSE

Copyright (c) 2006-2009 under the same (always latest) license(s) used by the Perl 6 /src 
branch of the Pugs trunk.

=head1 Table index

POD

	my @lines = split /\n/, $res->content;
	my %help = ();
	foreach my $line (@lines) {
		chomp $line;
		if($line =~ /<li><strong>(.+?)<\/strong>(.+?)<\/li>/) {
			my ($item, $item_description)= (_to_text($1), _to_text($2) );
                        $item_description =~ s/^\s+//;
			if($help{$item}) {
				$help{$item} .= "\n=item $item_description\n";
			} else {
				$help{$item} = "=item $item_description\n";
			}
		}
	}

	foreach my $item (sort keys %help) {
		$pod .=  "=head2 C<<< $item >>>\n\n=over\n\n" . $help{$item} . "\n=back\n\n";
	}
	
	return $pod;
}

my $pod = _perl6_table_index_to_pod 'http://www.perlfoundation.org/perl6/index.cgi?perl_6_index_tablet';
my $filename = 'perl6_table_index.pod';
print "Writing to $filename\n";
open FILE, ">$filename" or die "Cannot open $filename for writing\n";
print FILE $pod;
close FILE;

print "\n\n----- FINISHED -----\n\n";

# Copyright 2008-2009 Ahmad M. Zawawi and Gabor Szabo.
# LICENSE
# This program is free software; you can redistribute it and/or
# modify it under the same terms as Perl 5 itself.


lib/Perl6/Doc.pm
package Perl6::Doc;
use strict;
use warnings FATAL => 'all';

1;

=encoding utf8

=head1 NAME

Perl6::Doc - Perl 6 documentation

=head1 SYNOPSIS

This distribution is documentation-only. It contains no code for you to run.
Included are various Perl 5 Pod and Perl 6 Pod files describing the Perl 6
language. For a tool to read this documentation, see L<grok|grok>.

=head1 DESCRIPTION

Currently included in this distribution:

=head2 Design docs

=head3 Apocalypses (outdated)

Larry Wall started the Apocalypse (latin for revelation) series as a
systematic way of answering the RFCs (Request For Comments) that
started the design process for Perl 6.  Each Apocalypse corresponds to
a chapter in the book I<Programming Perl, 3rd edition>, and addresses
the features relating to that chapter in the book that are likely to
change.

Larry addresses each relevant RFC, and gives reasons why he accepted
or rejected various pieces of it.  But each Apocalypse also goes
beyond a simple "yes" and "no" response to attack the roots of the
problems identified in the RFCs.

B<Note:> These documents are outdated and and will not be updated to match
the specification (see L</Synopses>).

=head3 Exegeses (outdated)

Damian Conway's Exegeses (latin for I<explanation>) are extensions of
each Apocalypse.  Each Exegesis is built around a practical code
example that applies and explains the new ideas.

B<Note:> These documents are outdated and and will not be updated to match
the specification (see L</Synopses>).

=head3 Synopses

The Synopsis (latin for comparison) started as a fast to read diff
between Perl 5 and 6. Because they are also easier to maintain, all
changes of the language, that are evolving from the design process
are written down here first. The Apocalypses and Exegeses are frozen
as "historic documents".

In other words, these docs may change slightly or radically. But the
expectation is that they are "very close" to the final shape of Perl 6.

The Synopsis documents are to be taken as the formal specification for
Perl 6 implementations, while still being reference documentation for
Perl 6, like I<Programming Perl> is for Perl 5.

Note that while these documents still being subjected to the rigours 
of cross-examination through implementation.

=head2 Magazine articles

These are Pod-ified versions of magazine articles about Perl 6 that have
appeared in print during the design process. Take these with a grain of salt
as they might not describe Perl 6 as it is now. They are mostly from
L<http://www.perl.com> and L<http://www.perl-magazin.de/>.

=head2 Perl 6 Table Index

A Pod-ified version of L<http://www.perlfoundation.org/perl6/index.cgi?perl_6_index_tablet>
is included.

=head2 Man pages

Currently included are draft versions of F<perlintro> and F<perlsyn>.

=head1 HISTORY

As of version 0.40, this distribution has been overhauled and is no longer
maintained by Herbert Breunung, who originally took it over from ingy and
audreyt.

=head1 PACKAGING

Hinrik Örn Sigurðsson, L<hinrik.sig@gmail.com>

I'm not the author of the documentation in this distro, of course, I merely
maintain this package.

=head1 SOURCES

All Apocalypses and Exegeses were taken from the official Perl development
site: L<http://dev.perl.org/perl6/>

All Synopese were taken from the Pugs repository:
L<http://svn.pugscode.org/pugs/docs/Perl6/>

The magazine articles are from perl.com:
L<http://perl.com/pub/q/Article_Archive#Perl%206>

=head1 LICENSE AND COPYRIGHT

This copyright applies only to the C<Perl6::Doc> Perl software
distribution, not the documents bundled within.

Copyright (c) 2009, Hinrik Örn Sigurðsson L<hinrik.sig@gmail.com>.

C<grok> is distributed under the terms of the Artistic License 2.0.
For more details, see the full text of the license in the file F<LICENSE>
that came with this distribution.

=head2 Scribes

These are the authors of the included docs, named in the order their work 
was added. This list is not exhaustive.

* Larry Wall L<larry@wall.org>

* Damian Conway L<damian@conway.org>

* Luke Palmer L<luke@luqui.org>

* Allison Randal L<al@shadowed.net>

* Audrey Tang L<autrijus@cpan.org>

* Ingy döt Net L<ingy@cpan.org>

* Sam Vilain L<samv@cpan.org>

* Kirrily "Skud" Robert L<skud@cpan.org>

* Moritz Lenz L<moritz@fau2ik3.org>

* David Koenig L<karhu@u.washington.edu>

* Jonathan Scott Duff L<duff@pobox.com>

* Phil Crow L<philcrow2000@yahoo.com>

* chromatic L<chromatic@oreilly.com>

* Mark-Jason Dominus L<mjd@songline.com>

* Shmarya L<shmarya.rubenstein@gmail.com>

* Pawel Murias L<13pawel@gazeta.pl>

* Herbert Breunung L<lichtkind@cpan.org>

=cut
